plugins/modules/api.py

```diff
@@ -365,6 +366,11 @@ def list_to_dic(self, ldict):
                 dict[p[0]] = p[1]
         return dict
 
+    def split_params(self, params):
+        if not isinstance(params, str):
+            self.errors('Parameters can only be a string, received %s' % type(params))
```

Comment:
```suggestion
```
Either this should result in an `AssertionError` raised (since it is an internal error, and not a user-facing error), or this shouldn't be checked at all.

Reply:
Fixed in ec0c4a995bb8bb0746d01364bd5ee6892e8a2ece.